components/file_system/src/lib.rs

```diff
@@ -25,26 +29,28 @@ use std::sync::{Arc, Mutex};
 use openssl::error::ErrorStack;
 use openssl::hash::{self, Hasher, MessageDigest};
 
-#[derive(Debug)]
+#[derive(Clone, Copy, Debug)]
 pub enum IOOp {
     Read,
     Write,
 }
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq)]
 pub enum IOType {
-    Other,
-    Read,
-    Write,
-    Coprocessor,
-    Flush,
-    Compaction,
-    Replication,
-    LoadBalance,
-    Import,
-    Export,
+    Other = 0,
+    Read = 1,
+    Write = 2,
+    Coprocessor = 3,
+    Flush = 4,
+    Compaction = 5,
+    Replication = 6,
+    LoadBalance = 7,
+    Import = 8,
+    Export = 9,
 }
 
+pub const IO_TYPE_VARIANTS: usize = 10;
```

Comment:
proc macro can help you do this, see https://docs.rs/variant_count/1.0.0/variant_count/